library/core/src/num/mod.rs

```diff
@@ -969,9 +970,10 @@ pub enum FpCategory {
 }
 
 #[doc(hidden)]
-trait FromStrRadixHelper: PartialOrd + Copy {
-    fn min_value() -> Self;
-    fn max_value() -> Self;
+trait FromStrRadixHelper:
+    PartialOrd + Copy + Add<Output = Self> + Sub<Output = Self> + Mul<Output = Self>
```

Comment:
Oh, this is great!  Not being unsafe is a good plan -- and it makes sense that the optimizer couldn't really do anything different with it anyway, vs just using these which will wrap, and the real win is not needing to check the overflow flag ever iteration.